kasa/xortransport.py

```diff
@@ -0,0 +1,225 @@
+"""Module for the XorTransport."""
+import asyncio
+import contextlib
+import errno
+import logging
+import socket
+import struct
+from pprint import pformat as pf
+from typing import Dict, Generator, Optional
+
+# When support for cpython older than 3.11 is dropped
+# async_timeout can be replaced with asyncio.timeout
+from async_timeout import timeout as asyncio_timeout
+
+from .deviceconfig import DeviceConfig
+from .exceptions import RetryableException, SmartDeviceException
+from .json import loads as json_loads
+from .protocol import BaseTransport
+
+_LOGGER = logging.getLogger(__name__)
+_NO_RETRY_ERRORS = {errno.EHOSTDOWN, errno.EHOSTUNREACH, errno.ECONNREFUSED}
+_UNSIGNED_INT_NETWORK_ORDER = struct.Struct(">I")
+
+
+class XorTransport(BaseTransport):
```

Comment:
```suggestion
class LegacyTransport(BaseTransport):
```
Maybe? Considering it has nothing to do with xor except the encryption.

Reply:
Sure, will think of a better name or use Legacy

Reply:
Let's keep the name as discussed, it's fine as it is.